contracts/vending-minter/src/contract.rs

```diff
@@ -82,8 +82,14 @@ pub fn instantiate(
         });
     }
 
+    // sanitize base token uri
+    let base_token_uri = msg
+        .init_msg
+        .base_token_uri
+        .trim()
+        .replacen("IPFS", "ipfs", 1);
```

Comment:
I don't think you can lowercase not sure if the ipfhash is canse sensitive.

`Url::parse(&base_token_uri)?;` will give you the scheme sanitized (ipfs lowercase), maybe it can just reconstruct it `parsed_token_uri.to_string()` ?

Reply:
Something like
```
 let mut base_token_uri = msg
        .init_msg
        .base_token_uri
        .trim();

let parsed_token_uri = Url::parse(&base_token_uri)?;
if parsed_token_uri.scheme() != "ipfs" {
        return Err(ContractError::InvalidBaseTokenURI {});
}
base_token_uri = parsed_token_uri.to_string();
```

Reply:
ipfs hashing uses sha-256 by default which is case sensitive https://docs.ipfs.tech/concepts/content-addressing/#what-is-a-cid

Reply:
`Url::parse` + `.scheme` is how I did it before. Wondering why we are doing it again: https://github.com/public-awesome/launchpad/blob/35ff8471fa6d62e95d3856c84e6aacb835da96a0/contracts/vending-minter/src/contract.rs#L86

Reply:
this is only the validation `Url::parse` will not fail because sanitizes the string being passed so `parsed_token_uri.scheme == 'ipfs'` returns true, but when we save the sate we still use what was passed in the message https://github.com/public-awesome/launchpad/blob/35ff8471fa6d62e95d3856c84e6aacb835da96a0/contracts/vending-minter/src/contract.rs#L149 storing whatever was passed.